tests/utils.py

```diff
@@ -98,3 +98,21 @@ def assert_function_in_collapsed(function_name: str, collapsed: Mapping[str, int
     assert any(
         (function_name in record) for record in collapsed.keys()
     ), f"function {function_name!r} missing in collapsed data!"
+
+
+def get_python_version(application_docker_container: Container):
+    if application_docker_container is not None:
+        exit_code, output = application_docker_container.exec_run(cmd="python --version")
+        if exit_code != 0:
+            return None
+    else:
+        r, w = os.pipe()
+        try:
+            subprocess.run("python --version".split(), check=True, stderr=w, stdout=w)
+            output = os.read(r, 32)
+        finally:
+            os.close(r)
+            os.close(w)
```

Comment:
can just use `subprocess.check_output`

Reply:
and add a comment that the python app is running on the host, so that's why it can be extracted that way

Reply:
> and add a comment that the python app is running on the host, so that's why it can be extracted that way

Pretty self explanatory, but OK